### ISSUE TITLE
Some feature ideas

### ISSUE THREAD

**Issue by BookWyrm114** [Renardjojo/PetForDesktop#67]
------------------------------
UI
------------------------------
This is probably already planned eventually, but a UI to manage settings, import pets, and manage multiple pets (when that becomes a feature), could be really cool
------------------------------
Better collision detection on desktop and websites
------------------------------
Browsers - detect images, text, and other web elements for collision instead of using screenshots, and use this to allow pets to stay on surfaces when scrolling

Windows/OS-related - ignore files, shortcuts, folders, etc. (or give them square collision and have collision be optional; ignore taskbar elements, only give collision to top of taskbar (or side/bottom if users have taskbar in different positions), allowing a pet to walk around on/in the taskbar, but not end up standing on taskbar icons and elements. Also allow pets to stick to windows when resizing/moving them.

Wallpaper - allow users to provide an image that Pets will use for wallpaper collision, transparent pixels will be nothing, black will be solid, and other elements could eventually be added as well (blue defining "water" areas, where pets can jump in and swim around, with a little splash animation played on the surface when entering/exiting, red defining areas that don't have collision, but that pets will avoid, green could define "houses" with connected brown pixels defining "entrances", and white pixels defining "windows", so pets can appear to enter houses or other spaces in the wallpaper, and appear correctly hidden from view, etc.)
------------------------------
Pets
------------------------------
- Allow pets to interact with the cursor by occasionally jumping at it, and, if they reach the cursor, hanging onto it for a bit
- small physics objects for pets to play with
- names for pets with little toggleable nametags
- 3d pet support, not sure how complicated this would be to implement (I assume it would probably be pretty complex), but support for a common file type like .gltf, .obj, or .fbx files would probably be good, and allow for easy creation of 3d pets.
- allow pets to move on/ interact with the sides/top of monitors and windows
- flying pets
- recycle bin/folder interaction: let pets sometimes jump in the recycle bin or folders and sometimes peek out until leaving (they stay in for a maximum of 30 secs, and opening the folder/recycle bin send the pet to that window, and recycling the bin with the pet inside makes the pet jump out, do a little "anger" animation and actively start jumping at/attacking the cursor for the next minute or so)
------------------------------

Also side note: This app is incredible and I can't wait to see how it continues to improve, the collision on wallpapers and other elements is really unique and very well-done, hopefully you make the patreon page soon.

Also thanks for making it through this wall of text

**Comment by Renardjojo** [comment]
Hi, thank you very much for being the first user to send me feedback from github. It's a huge reward after all the work we've done to see that people are enjoying this application.

About the next update:
You can follow the next update of the application [here](https://github.com/Renardjojo/PetForDesktop/milestone)
You can also follow my page [discord](https://discord.com/invite/zuBup4N8) to receive all the changes and updates live. Currently there is nobody but all applications start somewhere with nobody.

Regarding your comments:
I have no intention of changing the collision detection system. It allows me to avoid tracking all the texts, images, applications... and creating a lot of extra work for each of them, as well as portability issues. I plan to run it on Linux one day and I'm trying to maintain a flexible architecture.
I do plan to add more interactions like a ball to catch, a home for our pets and the ability to pet them. For the moment, I'm struggling with some aspects of the window to make this application as discreet and light as possible. 
I've also created a patroon page but haven't yet configured it effectively for the community. The patroon page is planned for this stage and should be available in the next update.
For 3D pets, this is possible and relatively easy to do but will require more work. It could be a fun feature to do in the future but this application needs a lot of intnerraction. I'll have to concentrate on it after the engine refactoring I'm doing.

Once again, thank you very much for your comments. There are indeed a lot of things to do with this application, such as the user interface, interaction and portability.
I hope to release a new update soon to continue improving the user experience of this application! :)
Thanks and happy reading! 🦊
Jo

**Comment by Renardjojo** [comment]
I created the Patreon page: https://www.patreon.com/PetForDesktop

**Comment by BookWyrm114** [comment]
thanks! it's great that you plan to get it running on linux at some point, and the fact that 3D pets could be a feature at some point is amazing, can't wait to see the next update! I'll need to join the discord at some point I completely forgot about that.

**Comment by Renardjojo** [comment]
Yes, I'm currently working on the next version and hope to release it this week with many new features to improve interaction and production speed.
Thank you for supporting this project !
